protocol/x/clob/keeper/orders.go

```diff
@@ -334,6 +329,12 @@ func (k Keeper) placeOrder(
 		return 0, 0, err
 	}
 
+	// Validate that adding the order wouldn't exceed subaccount equity tier limits.
+	err = k.ValidateSubaccountEquityTierLimitForNewOrder(ctx, order)
```

Comment:
For my understanding why not add this in `PerformStatefulOrderValidation`? Is it specifically to ensure this isn't called for Short-Term order validation in `DeliverTx`?